std/math.d

```diff
@@ -7703,14 +7719,20 @@ T nextafter(T)(const T x, const T y) @safe pure nothrow @nogc
     enum float a = 1;
     static assert(is(typeof(nextafter(a, a)) == float));
     static assert(nextafter(a, a.infinity) > a);
+    static assert(isNaN(nextafter(a, a.nan)));
+    static assert(isNaN(nextafter(a.nan, a)));
 
     enum double b = 2;
     static assert(is(typeof(nextafter(b, b)) == double));
     static assert(nextafter(b, b.infinity) > b);
+    static assert(isNaN(nextafter(b, b.nan)));
+    static assert(isNaN(nextafter(b.nan, b)));
 
-    //enum real c = 3;
-    //static assert(is(typeof(nextafter(c, c)) == real));
+    enum real c = 3;
+    static assert(is(typeof(nextafter(c, c)) == real));
```

Comment:
When I read #7176 correctly, this unittest was commented out on purpose by @n8sh, as it might not work on every real implementation yet.

Reply:
It doesn't matter in these special cases as long as `realX == realY` and `isNaN(realVal)` work properly at compile time. #7176 just mentions `nextUp` / `nextDown` for `real`, called if both of the arguments are different and not NaN.

Reply:
I didn't need to comment out those two lines since `typeof` does not require the contained expression to run in CTFE.